src/python/gudhi/representations/kernel_methods.py

```diff
@@ -125,7 +125,7 @@ def transform(self, X):
         Returns:
             numpy array of shape (number of diagrams in **diagrams**) x (number of diagrams in X): matrix of pairwise sliced Wasserstein kernel values.
         """
-        return pairwise_persistence_diagram_kernels(X, self.diagrams_, kernel="sliced_wasserstein", bandwidth=self.bandwidth, num_directions=self.num_directions, n_jobs=self.n_jobs)
+        return pairwise_persistence_diagram_kernels(self.diagrams_, X, kernel="sliced_wasserstein", bandwidth=self.bandwidth, num_directions=self.num_directions, n_jobs=self.n_jobs)
```

Comment:
There is indeed an issue here, but it is in the doc: in scikit-learn pipelines, when computing new kernel matrices on, say, some test data (usually provided as the `X` argument), the shape of the matrix should be [new_test_data, train_data]. Hence, we should keep `X, self.diagrams_` as first two arguments but change the doc to `(number of diagrams in X) x (number of diagrams in **diagrams**)`

Reply:
Oh, thank you for pointing that out! I reverted the transposition and I have changed the docs for all transformers in `src/python/gudhi/representations/kernel_methods`.